testsuite/tests/lib-marshal/intern_final.ml

```diff
@@ -0,0 +1,30 @@
+(* TEST *)
+
+let t : int array = Array.make 200 42
+let c = open_out_bin "data42"
+let () = Marshal.to_channel c t []
+let () = close_out c
+
+let t : int array = Array.make 200 0
+let c = open_out_bin "data0"
+let () = Marshal.to_channel c t []
+let () = close_out c
```

Comment:
This would not fail, because the code for unmarshalling bytes already restores the values of the global variables. It does so for a different reason: the reason is that `input_src` then points to the global heap, and the GC might move the source data.

Reply:
What's the issue with files in tests? OCamltest uses a specific directory for each tests, which can easily be removed. Also, note that the compilation artifacts always exist.